frontend/src/features/public-form/PublicFormService.ts

```diff
@@ -152,16 +154,37 @@ export const submitFormFeedback = async (
 }
 
 /**
- * Obtain payment receipt for a given submission.
+ * Obtain payment receipt status for a given submission.
  * @param formId the id of the form
  * @param submissionId the id of the form submission
- * @returns success message
+ * @returns PaymentReceiptStatusDto on success
  */
-export const getPaymentReceipt = async (
+export const getPaymentReceiptStatus = async (
   formId: string,
   submissionId: string,
-): Promise<PaymentReceiptDto> => {
-  return ApiService.get<PaymentReceiptDto>(
-    `payments/receipt/${formId}/${submissionId}`,
+): Promise<PaymentReceiptStatusDto> => {
+  return ApiService.get<PaymentReceiptStatusDto>(
+    `payments/receipt/${formId}/${submissionId}/status`,
   ).then(({ data }) => data)
 }
+
+/**
+ * Downloads payment receipt and save to disk in pdf form
+ * @param formId the id of the form
+ * @param submissionId the id of the form submission
+ * @returns void
+ */
+export const downloadPaymentReceipt = async (
+  formId: string,
+  submissionId: string,
+): Promise<void> => {
+  return ApiService.get(
+    `payments/receipt/${formId}/${submissionId}/download`,
+  ).then(({ data }) => {
+    const buffer = Buffer.from(data.pdfBuffer.data)
+
+    const blob = new Blob([buffer], { type: 'application/pdf' })
+    FileSaver.saveAs(blob, `${submissionId}-receipt.pdf`)
+    return
+  })
+}
```

Comment:
I don't think we need to handle this ourselves. We can leverage the browser's native capabilities to download a file by using the [Content-Disposition header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition) and not have to hold anything in memory. The `Download` button should just act as link to make it happen.

There is additional an issue with separation of concerns here 🤔. The file is generated as pdf on the server, so the server should set a content-type header and the appropriate file name+extension in the response.